hazelcast/src/main/java/com/hazelcast/map/impl/wan/WanMapEntryView.java

```diff
@@ -0,0 +1,252 @@
+
+    private final K key;
+    private final V value;
+    private final long cost;
+    private final long creationTime;
+    private final long expirationTime;
+    private final long hits;
+    private final long lastAccessTime;
+    private final long lastStoredTime;
+    private final long lastUpdateTime;
+    private final long version;
+    private final long ttl;
+
+    public WanMapEntryView(EntryView<K, V> entryView) {
+        this.key = entryView.getKey();
+        this.value = entryView.getValue();
+        this.cost = entryView.getCost();
+        this.version = entryView.getVersion();
+        this.hits = entryView.getHits();
+        this.lastAccessTime = entryView.getLastAccessTime();
+        this.lastUpdateTime = entryView.getLastUpdateTime();
+        this.ttl = entryView.getTtl();
+        this.creationTime = entryView.getCreationTime();
+        this.expirationTime = entryView.getExpirationTime();
+        this.lastStoredTime = entryView.getLastStoredTime();
+    }
+
+    @Override
+    public K getKey() {
+        return key;
+    }
+
+    @Override
+    public V getValue() {
+        return value;
+    }
+
+    @Override
+    public long getCost() {
+        return cost;
+    }
+
+    @Override
+    public long getCreationTime() {
+        return creationTime;
+    }
+
+    @Override
+    public long getExpirationTime() {
+        return expirationTime;
+    }
+
+    @Override
+    public long getHits() {
+        return hits;
+    }
+
+    @Override
+    public long getLastAccessTime() {
+        return lastAccessTime;
+    }
+
+    @Override
+    public long getLastStoredTime() {
+        return lastStoredTime;
+    }
+
+    @Override
+    public long getLastUpdateTime() {
+        return lastUpdateTime;
+    }
+
+    @Override
+    public long getVersion() {
+        return version;
+    }
+
+    @Override
+    public long getTtl() {
+        return ttl;
+    }
+
+    @Override
+    public long getMaxIdle() {
+        return 0;
+    }
+
+    /**
+     * Needed for client protocol compatibility.
+     */
+    @SuppressWarnings("unused")
+    public long getEvictionCriteriaNumber() {
+        return 0;
+    }
+
+    /**
+     * Needed for client protocol compatibility.
+     */
+    @SuppressWarnings("unused")
+    public void setEvictionCriteriaNumber(long evictionCriteriaNumber) {
+    }
+
+    @Override
+    public void writeData(ObjectDataOutput out) throws IOException {
+        IOUtil.writeObject(out, key);
+        IOUtil.writeObject(out, value);
+        out.writeLong(cost);
+        out.writeLong(creationTime);
+        out.writeLong(expirationTime);
+        out.writeLong(hits);
+        out.writeLong(lastAccessTime);
+        out.writeLong(lastStoredTime);
+        out.writeLong(lastUpdateTime);
+        out.writeLong(version);
+        // writes the deprecated evictionCriteriaNumber to the data output (client protocol compatibility)
```

Comment:
The "client protocol compatibility" part in the comment is misleading (carried over from `SimpleEntryView`?). A separate [`EntryViewCodec`](https://github.com/hazelcast/hazelcast-client-protocol/blob/9df8ac22efe2a186505de0b0c6a2af15e564e883/hazelcast/src/main/java/com/hazelcast/client/impl/protocol/codec/EntryViewCodec.java#L27) is used when transferring `SimpleEntryView` to/from clients, the `IdentifiedDataSerializable` serialized form is only used across members.

Reply:
Copy-paste, changed the comment here.